Sming/Components/crypto/include/Crypto/Md5.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "HashContext.h"
+
+namespace Crypto
+{
+using Md5Hash = HashValue<MD5_SIZE>;
+
+class Md5 : public HashContext<Md5, Md5Hash>
+{
+public:
+	using Hash = Md5Hash;
+
+	Md5()
+	{
+		init();
+	}
+
+	void init()
```

Comment:
Appears redundant to constructor.
Instead of
```cpp
Md5 md5;
...
md5.init()
```
the user could simply write
```
Md5 md5;
...
md5 = Md5();
```

Reply:
I figure the `init`, `update` and `final` sequence is more familiar to existing crypto users.

Reply:
Yeay, but in C++, it's usually the constructors job to do the 'init'. It creates ambiguity if it is always necessary to call `init` after construction or only for reused objects.

Reply:
I see your point. With the 'Engine' revision it makes less sense keeping init, so I'll remove it.

Reply:
I'm going to add a method back, but called `reset()` to resolve the ambiguity.

It makes the code very clunky otherwise, for example:

```
// Reset (init) on construction
Crypto::Sha1 certificateSha1;

// (1) Reset by assignment: need to specify the type again
certificateSha1 = Crypto::Sha1();

// (2) Reset by method: don't care about the type
certificateSha1.reset();
```

Using (2) means we can change the hash algorithm simply by amending the declared type, everything else remains the same.

Reply:
That's a much better name for the method. Now it adds value over the constructor.

Reply:
FYI: One could use brace initialization to avoid repeating the type:
```
certificateSha1 = {};
```
However, `reset()` is clearly more expressive in this case.